### PR TITLE
Rename bounding ‘box’ to ‘rect’; move structure to geo-types.

### PR DESCRIPTION
**Pull request by frewsxcv**
- Rename `Bbox` structure to `Rect`, move to `geo_types`, transition
  from four `T` values to two `Coordinate<T>` values
- Rename `BoundingBox` to `BoundingRect`

Fixes https://github.com/georust/rust-geo/issues/292.

### CODE REVIEW

File: geo/src/algorithm/bounding_rect.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+    ///
+    /// Return the BoundingRect for a MultiPoint
+    ///
+    fn bounding_rect(&self) -> Self::Output {
+        get_bounding_rect(self.0.iter().map(|p| p.0))
+    }
+}
+
+impl<T> BoundingRect<T> for Line<T>
```
Comment (michaelkirk) [inline geo/src/algorithm/bounding_rect.rs]: Are the Line and LineString implementations intentionally duplicated in geo-types/src/algorithms.rs?

File: geo-types/src/line_string.rs [hunk 2]
```diff
@@ -174,7 +174,7 @@ where
     type Point = Point<T>;
 
     fn mbr(&self) -> ::spade::BoundingRect<Self::Point> {
-        let bbox = self.bbox();
+        let bbox = self.bounding_rect();
```
Comment (michaelkirk) [inline geo-types/src/line_string.rs]: nit: seems like you're preferring `let bounding_rect =` elsewhere

Comment (frewsxcv) [reply geo-types/src/line_string.rs]: yep, good catch, thanks

File: geo/src/algorithm/bounding_rect.rs (hunk 1 above)
Comment (frewsxcv) [reply geo/src/algorithm/bounding_rect.rs]: yeah, intentional. to have `spade` trait implementations for the types in `geo-types`, we need to have some basic algorithms included in the `geo-types` crate. but i'm planning to consolidate them in a follow-up pull request to this one.

**Pull request merged**